packages/nextjs/app/page.tsx

```diff
@@ -2,48 +2,106 @@
 
 import Link from "next/link";
 import type { NextPage } from "next";
-import { BugAntIcon, MagnifyingGlassIcon } from "@heroicons/react/24/outline";
+import { BugAntIcon, MagnifyingGlassIcon, RocketLaunchIcon, UserGroupIcon } from "@heroicons/react/24/outline";
+import BuildersGrid from "~~/components/batch/BuildersGrid";
+import { useScaffoldReadContract } from "~~/hooks/scaffold-eth/useScaffoldReadContract";
 
 const Home: NextPage = () => {
+  const {
+    data: checkedInCount,
+    isLoading,
+    error,
+  } = useScaffoldReadContract({
+    contractName: "BatchRegistry",
+    functionName: "checkedInCounter",
+  });
+
+  const displayCount = isLoading ? (
+    <span className="loading loading-spinner loading-sm text-primary"></span>
+  ) : error ? (
+    "Error loading"
+  ) : typeof checkedInCount === "bigint" ? (
+    checkedInCount.toString()
+  ) : (
+    "0"
+  );
+
```

Comment:
This is out of scope and relates to issue #6. Please remove it.

Reply:
Please remove this. It belongs to issue #6. If you want to include the "checkedInCounter", create a separate PR for it.

In a "real-world" scenario, when working on an issue, ensure you only add code that falls within its scope.